src/main/java/reactor/rabbitmq/Receiver.java

```diff
@@ -209,16 +230,27 @@ protected void completeOnChannelShutdown(Channel channel, FluxSink<?> emitter) {
     // TODO consume with dynamic QoS and/or batch ack
 
     public void close() {
-        if (hasConnection.getAndSet(false)) {
-            try {
-                // FIXME use timeout on block (should be a parameter of the Receiver)
-                connectionMono.block().close();
-            } catch (IOException e) {
-                throw new RabbitFluxException(e);
+        if (closingOrClosed.compareAndSet(false, true)) {
+            if (hasConnection.getAndSet(false)) {
+                safelyExecute(
+                        () -> connectionMono.block().close(this.connectionClosingTimeout),
```

Comment:
If the connectionMono has never been subscribed, or if the original connection attempted failed (which is no longer cached), then connectionMono.block() will trigger a new connection attempt.

I'm wondering if [keeping track of the connections](https://gist.github.com/philsttr/9a330dda180047368abce3a360da45a7#file-fooqueueprocessor-java-L239-L242), and then [closing them here](https://gist.github.com/philsttr/9a330dda180047368abce3a360da45a7#file-fooqueueprocessor-java-L434) would be a better approach.

Reply:
Yes, makes sense. The connection is now tracked with a `AtomicReference<Connection>`.